app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -365,14 +371,9 @@ private void getFavorFromView(FavorStatus status) {
     // Upload picture to database if it exists
     if (mImageView.getDrawable() != null) {
       Bitmap picture = ((BitmapDrawable) mImageView.getDrawable()).getBitmap();
-
-      // TODO: display result of uploading picture somewhere
-      CompletableFuture<String> pictureUrl = PictureUtil.uploadPicture(picture);
-      pictureUrl.thenAccept(url -> FavorUtil.getSingleInstance().updateFavorPhoto(favor, url));
-      pictureUrl.exceptionally(e -> {
-        // TODO: create UI element that informs the user that the picture wasn't uploaded
-        return null;
-      });
+      getViewModel().uploadOrUpdatePicture(favor, picture);
```

Comment:
handle exceptional result (as you did before)
To show a snackbar you can call the CommonTools.showSnackbar(requireView(),getString(R.string.anyerrormessagehere))